internal/tiltfile/docker_compose.go

```diff
@@ -37,33 +40,77 @@ type dcResourceSet struct {
 func (dc dcResourceSet) Empty() bool { return reflect.DeepEqual(dc, dcResourceSet{}) }
 
 func (s *tiltfileState) dockerCompose(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	configPaths := value.NewLocalPathListUnpacker(thread)
+	var configPaths starlark.Value
 
 	err := s.unpackArgs(fn.Name(), args, kwargs, "configPaths", &configPaths)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, v := range configPaths.Value {
-		err = io.RecordReadPath(thread, io.WatchFileOnly, v)
-		if err != nil {
-			return nil, err
-		}
+	paths := starlarkValueOrSequenceToSlice(configPaths)
+
+	if len(paths) == 0 {
+		return nil, fmt.Errorf("Nothing to compose")
 	}
 
 	dc := s.dc
+
 	currentTiltfilePath := starkit.CurrentExecPath(thread)
 	if dc.tiltfilePath != "" && dc.tiltfilePath != currentTiltfilePath {
 		return starlark.None, fmt.Errorf("Cannot load docker-compose files from two different Tiltfiles.\n"+
 			"docker-compose must have a single working directory:\n"+
 			"(%s, %s)", dc.tiltfilePath, currentTiltfilePath)
 	}
 
-	// To make sure all the docker-compose files are compatible together,
-	// parse them all together.
-	allConfigPaths := append([]string{}, dc.configPaths...)
-	allConfigPaths = append(allConfigPaths, configPaths.Value...)
-	project := model.DockerComposeProject{ConfigPaths: allConfigPaths}
+	project := model.DockerComposeProject{
+		ConfigPaths: dc.configPaths,
+		ProjectPath: dc.Project.ProjectPath,
+		Name:        model.NormalizeName(filepath.Base(currentTiltfilePath)),
+	}
+
+	for _, val := range paths {
+		switch v := val.(type) {
+		case nil:
+			continue
+		case io.Blob:
+			yaml := v.String()
+			message := "unable to store yaml blob"
+			tmpfile, err := os.CreateTemp("", fmt.Sprintf("%s-docker-compose-*.yml", filepath.Base(filepath.Dir(currentTiltfilePath))))
+			if err != nil {
+				return nil, errors.Wrap(err, message)
+			}
+			_, err = tmpfile.WriteString(yaml)
+			if err != nil {
+				return nil, errors.Wrap(err, message)
+			}
+			err = tmpfile.Close()
+			if err != nil {
+				return nil, errors.Wrap(err, message)
+			}
+			project.ConfigPaths = append(project.ConfigPaths, tmpfile.Name())
+		default:
+			path, err := value.ValueToAbsPath(thread, val)
+			if err != nil {
+				return starlark.None, fmt.Errorf("expected blob | path (string). Actual type: %T", val)
+			}
+
+			// Set project path to dir of first compose file, like DC CLI does
+			if project.ProjectPath == "" {
+				project.ProjectPath = filepath.Dir(path)
```

Comment:
I think this should also be where `project.Name` is set - currently it's going to name after the `Tiltfile` dir

Reply:
Yeah, since we only support one compose spec per Tilt instance right now, I made the specific decision to always name the project after the directory the Tiltfile is in. We could do this once we support multiple compose specs per instance...

Reply:
Yeah, I think a follow-on PR to add some more of the DC command-line options -- eg. #5265 -- is a strong possibility.